### ISSUE TITLE
Calculate diffuse fraction before IAM and spectral correction

### ISSUE THREAD

**Issue by williamhobbs** [williamhobbs/pv-system-model#26]
Because diffuse fraction (`fd`) is calculated after IAM and spectral correction is applies to global POA (`poa_total_without_direct_shade`), it's possible to have a diffuse fraction greater than 1 and then an incorrect (possible negative) shading loss. 

`fd` needs to be calculated from global POA before those corrections are applied. 

https://github.com/williamhobbs/pv-system-model/blob/7d12044b61426e8768651894d1e32af758c580c3/pv_model.py#L493-L502

for example (note use of `fd.values` instead of `fd` in later calcs),

```python
    # diffuse fraction
    fd = (total_irrad['poa_diffuse'] / total_irrad['poa_global'])

    # calculate shade loss for each course/string
    if shade_loss_model == 'linear':
        shade_loss = fs * (1 - fd.values)
    elif (shade_loss_model == 'non-linear_simple' or
          shade_loss_model == 'non-linear_simple_twin_module'):
        shade_loss = non_linear_shade(n_cells_up, fs, fd.values)
```
